packages/machine/src/protocol/utils/signature-validator.ts

```diff
@@ -1,26 +1,21 @@
-import { recoverAddress } from "ethers/utils";
+import { recoverAddress, Signature } from "ethers/utils";
 
-import { StateChannel } from "../../models";
-import { ProtocolMessage } from "../../protocol-types-tbd";
-import { Context } from "../../types";
+import { EthereumCommitment } from "../../ethereum/types";
 
 export function validateSignature(
-  message: ProtocolMessage,
-  context: Context,
-  state: StateChannel
+  expectedSigner: string,
+  commitment?: EthereumCommitment,
+  signature?: Signature
```

Comment:
Why is this optional? The function shouldn't even be called if a signature can't be given for validation.

Reply:
To reduce code reuse in the protocol flow files and this file.

Reply:
Doesn't sound like a clean separation but non-blocking for now.